lib/Service/TemplateFieldService.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Richdocuments\Service;
+
+use OCA\Richdocuments\AppConfig;
+use OCP\Files\IRootFolder;
+use OCP\Files\Node;
+use OCP\Files\Template\Field;
+use OCP\Files\Template\FieldType;
+use OCP\Http\Client\IClientService;
+
+class TemplateFieldService {
+	private IClientService $clientService;
+	private AppConfig $appConfig;
+	private IRootFolder $rootFolder;
+
+	public function __construct(
+		IClientService $clientService,
+		AppConfig $appConfig,
+		IRootFolder $rootFolder
+	) {
+		$this->clientService = $clientService;
+		$this->appConfig = $appConfig;
+		$this->rootFolder = $rootFolder;
+	}
+
+	/**
+	 * @param Node|int $file
+	 * @return array|string
+	 */
+	public function extractFields(Node|int $file) {
+		if (is_int($file)) {
+			$file = $this->rootFolder->getFirstNodeById($file);
+		}
+
+		$collaboraUrl = $this->appConfig->getCollaboraUrlInternal();
+		$httpClient = $this->clientService->newClient();
+
+		$formData = [
+			'name' => 'data',
+			'contents' => $file->getStorage()->fopen($file->getInternalPath(), 'r'),
+			'headers' => ['Content-Type' => 'multipart/form-data'],
+		];
+
+		try {
+			$response = $httpClient->post(
+				$collaboraUrl . "/cool/extract-document-structure",
+				[
+					'query' => ['limit' => 'content-control'],
+					'multipart' => [$formData],
+				]
```

Comment:
Requests should also pass the default options, @mejo- just yesterday wrapped that nicely in https://github.com/nextcloud/richdocuments/pull/3831/files#diff-c2d693a39b27f535396a48fdaad4a41a517c489be3a492fb968ac91e5e0181efR14-R21